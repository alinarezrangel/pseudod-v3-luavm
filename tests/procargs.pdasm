PDVM 1.0
PLATFORM "pdcrt"

SECTION "code"
  MK0CLZ 1
  MSG 0, 0, 1
  PRN
  NL

  ICONST 1
  MK0CLZ 1
  MSG 0, 1, 1
  PRN
  NL

  MK0CLZ 2
  MSG 0, 0, 0

  ICONST 1
  MK0CLZ 2
  MSG 0, 1, 0

  ICONST 1
  ICONST 2
  MK0CLZ 2
  MSG 0, 2, 1
  PRN
  NL

  ICONST 5
  ICONST 4
  MK0CLZ 3
  MSG 0, 2, 1
  PRN
  NL

  ICONST 5
  ICONST 4
  ICONST 100
  ICONST 1000
  MK0CLZ 3
  MSG 0, 4, 1
  PRN
  NL
ENDSECTION

SECTION "procedures"
  PROC 1
    ICONST 1
    RETN 1
  ENDPROC

  PROC 2
    PARAM 0
    LGET 0
    PRN
    NL
    RETN 0
  ENDPROC

  PROC 3
    PARAM 0
    PARAM 1
    LGET 0
    LGET 1
    MUL
    RETN 1
  ENDPROC
ENDSECTION

SECTION "constant pool"
  #0 STRING "llamar"
ENDSECTION
