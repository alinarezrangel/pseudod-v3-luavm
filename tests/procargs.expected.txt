1
1
NULO
1
1
NULO
20
20
